File: VisaCheckoutSDK/SampleApps/VisaCheckoutExampleObjectiveC/VisaCheckoutExampleObjectiveC/ViewController.m
/**
 Copyright © 2017 Visa. All rights reserved.
 */

#import "ViewController.h"
#import "VisaCheckoutButton+Designable.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet VisaCheckoutButton *checkoutButton;
@end

@implementation ViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    
    /// See the documentation/headers for `VisaPurchaseInfo` for
    /// various ways to customize the purchase experience.
    VisaCurrencyAmount *total = [[VisaCurrencyAmount alloc] initWithDouble:22.09];
    VisaPurchaseInfo *purchaseInfo = [[VisaPurchaseInfo alloc] initWithTotal:total
                                                                    currency:VisaCurrencyUsd];
    purchaseInfo.reviewAction = VisaReviewActionPay;
    
    [self.checkoutButton
     onCheckoutWithPurchaseInfo:purchaseInfo
     completion:^(VisaCheckoutResult *result) {
         switch (result.statusCode) {
             case VisaCheckoutResultStatusSuccess:
                 NSLog(@"Encrypted key: %@", result.encryptedKey);
                 NSLog(@"Payment data: %@", result.encryptedPaymentData);
                 break;
             case VisaCheckoutResultStatusUserCancelled:
                 NSLog(@"Payment cancelled by the user");
                 break;
             default:
                 break;
         }
     }];
}
@end

File: VisaCheckoutSDK/SampleApps/VisaCheckoutExampleSwift/VisaCheckoutExampleSwift/ViewController.swift
/**
 Copyright © 2017 Visa. All rights reserved.
 */

import UIKit
import VisaCheckoutSDK

class ViewController: UIViewController {
    @IBOutlet var checkoutButton: VisaCheckoutButton!
    override func viewDidLoad() {
        super.viewDidLoad()
        
        /// See the documentation/headers for `PurchaseInfo` for
        /// various ways to customize the purchase experience.
        let purchaseInfo = PurchaseInfo(total: 22.09, currency: .usd)
        purchaseInfo.reviewAction = .pay
        
        checkoutButton.onCheckout(purchaseInfo: purchaseInfo) { result in
            switch result.statusCode {
            case .success:
                print("Encrypted key: \(String(describing: result.encryptedKey))")
                print("Payment data: \(String(describing: result.encryptedPaymentData))")
            case .userCancelled:
                print("Payment cancelled by the user")
            default:
                break
            }
        }
    }
}
